src/expr.rs

```diff
@@ -1779,7 +1779,10 @@ fn rewrite_call_args(context: &RewriteContext,
     };
 
     match write_list(&item_vec, &fmt) {
-        Some(ref s) if !s.contains('\n') && s.len() > one_line_width => {
+        // If arguments do not fit in a single line and do not contain newline,
+        // try to put it on the next line.
+        Some(ref s) if context.config.fn_call_style == IndentStyle::Block &&
+                       (!s.contains('\n') && s.len() > one_line_width) => {
```

Comment:
So while I think this fix is necessary, I don't think it is sufficient - even in block mode, it is a bug to introduce a comma in a macro where there wasn't one before. I think we have a flag which indicates we're formatting a macro rather than a function and we should check that around here some where.